### PR TITLE
rec: Followup on ZoneToCache and fix #10246: document local root config

### LINKED ISSUE #10246: rec: document local root config (rfc8806)
Theres are now two ways to do this:

- use zoneToCache for . (#10505)
- use a forward to an auth that has the root zone (e.g. by axfr)

Both methods should be documented with config examples.

### PR DESCRIPTION
**Pull request by omoerbeek**
### Short description

Fixes #10246
### Checklist

I have:
- [X] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [X] compiled this code
- [X] tested this code
- [X] included documentation (including possible behaviour changes)
- [X] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)

### CODE REVIEW

File: pdns/recursordist/docs/lua-config/ztc.rst [hunk 1]
```diff
@@ -0,0 +1,75 @@
+.. _ztc:
+
+Zone to Cache
+-------------
+
+Zone to Cache is a function to load a zone to the Recursor cache periodically or once at startup.
```
Comment (rgacogne) [inline pdns/recursordist/docs/lua-config/ztc.rst:6]:
```suggestion
Zone to Cache is a function to load a zone into the Recursor cache periodically, or every time the Lua configuration is loaded, at startup and whenever ``rec_control reload-lua-config`` is issued.
```

File: pdns/recursordist/docs/lua-config/ztc.rst [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+This allows the Recursor to have an always hot cache for these zones.
+The zone to cache can be retrieved via zone transfer (AXFR format) or read from a zone file retrieved via http, https or a local file.
```
Comment (rgacogne) [inline pdns/recursordist/docs/lua-config/ztc.rst:8]:
```suggestion
The zone content to cache can be retrieved via zone transfer (AXFR format) or read from a zone file retrieved via http, https or a local file.
```

File: pdns/recursordist/docs/lua-config/ztc.rst [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+These options can be set in the ``settings`` of :func:`zoneToCache`.
+
+timeout
+~~~~~~~
+The maximum time (in seconds) a retrieval using the ``axfr`` or ``url`` method may take.
+Default is 20 seconds.
+
+tsigname
+~~~~~~~~
+The name of the TSIG key to authenticate to the server when using the ``axfr`` method
```
Comment (rgacogne) [inline pdns/recursordist/docs/lua-config/ztc.rst:44]:
```suggestion
The name of the TSIG key to authenticate to the server and validate the zone content with when using the ``axfr`` method.
```

File: pdns/recursordist/docs/lua-config/ztc.rst [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+When this is set, `tsigalgo`_ and `tsigsecret`_ must also be set.
+
+tsigalgo
+~~~~~~~~
+The name of the TSIG algorithm (like 'hmac-md5') used
```
Comment (rgacogne) [inline pdns/recursordist/docs/lua-config/ztc.rst:49]:
```suggestion
The name of the TSIG algorithm (like 'hmac-md5') used.
```

File: pdns/recursordist/docs/lua-config/ztc.rst [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+
+tsigsecret
+~~~~~~~~~~
+Base64 encoded TSIG secret
```
Comment (rgacogne) [inline pdns/recursordist/docs/lua-config/ztc.rst:53]:
```suggestion
Base64 encoded TSIG secret.
```

File: pdns/recursordist/docs/lua-config/ztc.rst [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+
+refreshPeriod
+~~~~~~~~~~~~~
+An integer describing the interval (in seconds) to wait between retrievals.
+A value of zero means the retrieval is done once at startup and on Lua configuration reload.
+By default, the refresh value is 86400 (24 hours).
```
Comment (rgacogne) [inline pdns/recursordist/docs/lua-config/ztc.rst:59]: Just a nit but a nice improvement would be to be able to use the refresh value provided in the SOA record.

**Review by rgacogne — APPROVED** [review]

File: pdns/recursordist/docs/lua-config/ztc.rst (hunk 6 above)
Comment (omoerbeek) [reply pdns/recursordist/docs/lua-config/ztc.rst:59]: Yes, this code will need an iteration anyway when zonemd will be implemented. I plan to do SOA based refresh then as well.

**Pull request merged**